File: plugins/capacitor-chromecast-master/ios/Plugin/ChromecastPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ChromecastPlugin, "Chromecast",
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestSession, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(loadMedia, CAPPluginReturnPromise);
)

File: plugins/capacitor-chromecast-master/ios/Plugin/ChromecastPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */

@objc(ChromecastPlugin)
public class ChromecastPlugin: CAPPlugin {
    private var implementation: Chromecast?

    @objc func initialize(_ call: CAPPluginCall) {
        DispatchQueue.main.sync {
            self.implementation = Chromecast();
            call.resolve();
        }
    }

    @objc func requestSession(_ call: CAPPluginCall) {
        implementation?.discoveryDevices(fn:{
            call.resolve([
                "conect": "ok"
            ]);
        });
        
    }

    @objc func loadMedia(_ call: CAPPluginCall) {
        implementation?.startStremming(url: call.getString("contentId")!, currentTime:call.getDouble("currentTime")!, poster: call.getString("poster")!, title: call.getString("title")!)
    }
}
